pathd/path_pcep_controller.c

```diff
@@ -612,8 +658,30 @@ int pcep_thread_event_handler(struct thread *thread)
 		ret = pcep_thread_event_update_pce_options(ctrl_state, pcc_id,
 							   pce_opts);
 		break;
+	case EV_UPDATE_BEST_PCE:
+		ret = pcep_thread_event_update_best_pce(ctrl_state, pcc_id);
+		break;
+	case EV_UPDATE_TIMER_BEST_PCE:
+		if (!pcc_id) {
+			// resync whatever were new best
+			pcc_id = calculate_best_pce(ctrl_state);
+			if (pcc_id) {
+				struct pcc_state *pcc_state =
+					get_pcc_state(ctrl_state, pcc_id);
+				pcep_thread_schedule_sync_best_pce(
+					ctrl_state, pcc_id, 30 /*delay*/,
+					&pcc_state->t_reconnect);
+			}
+		}
+		break;
 	case EV_REMOVE_PCC:
-		ret = pcep_thread_event_remove_pcc(ctrl_state, pcc_id);
+		pce_addr = (struct ipaddr *)payload;
```

Comment:
Here we should be sure we own the address, and free it.

Reply:
yes, done